gym/lib/gym/generators/package_command_generator_legacy.rb

```diff
@@ -47,7 +47,7 @@ def temporary_output_path
 
       def appfile_path
         path = Dir.glob("#{BuildCommandGenerator.archive_path}/Products/Applications/*.app").first
-        path ||= Dir[BuildCommandGenerator.archive_path + "/**/*.app"].last
+        path ||= Dir[BuildCommandGenerator.archive_path + "/**/*.app"].first
```

Comment:
I'm not sure why myself, I mainly switched it to first since it matches the first line, it seems very abitary. In this case we only want one result so I think first expresses that intent.

But thre are a few things like this across the tool we need to review.